### PR TITLE
feat: Add github service and related APIs

### LINKED ISSUE #31: Create github services APIs.
## Type of PR

- [x] Feature Request
- [ ] Report Bug
- [ ] Documentation Update
- [ ] Other

## Description

For creating one zap workflow, some APIs need to be supported. Also, store data which is needed to support then.

## Check
- [x] I have followed issue creation guidelines
- [x] I have attached proper label with the issue
- [x] The issue is assigned to a assignee

### PR DESCRIPTION
**Pull request by harish-aka-shivi**
## Description

It adds github service data, and adds the APIs to access it.

Fixes #31 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Test
- [ ] Integration Test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/src/controllers/serviceControllers.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const actionNames = actions.map(action => action.name);
+    res.send({ actions: actionNames });
+  } catch (error) {
+    res.status(500).send();
+  }
+};
+
+const getTriggerForService = async (req, res) => {
+  try {
+    const service = getAppropriateService(req.param('service'));
```
Comment (rishabh1403) [inline backend/src/controllers/serviceControllers.js]: `req.param('service')` are you calling a function here?

File: backend/src/thirdParties/github/githubService.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            login: 'Codertocat',
+            id: 21031067,
+            node_id: 'MDQ6VXNlcjIxMDMxMDY3',
+            avatar_url: 'https://avatars1.githubusercontent.com/u/21031067?v=4',
+            gravatar_id: '',
+            url: 'https://api.github.com/users/Codertocat',
+            html_url: 'https://github.com/Codertocat',
+            followers_url: 'https://api.github.com/users/Codertocat/followers',
+            following_url:
+              'https://api.github.com/users/Codertocat/following{/other_user}',
```
Comment (rishabh1403) [inline backend/src/thirdParties/github/githubService.js]: `'https://api.github.com/users/Codertocat/following{/other_user}'`
Can you please explain this? especially curly braces part

File: backend/src/controllers/serviceControllers.js (hunk 1 above)
Comment (harish-aka-shivi) [reply backend/src/controllers/serviceControllers.js]: Yes, this is a function. Nut now I re-read the express docs, I can use this using `req.params.service` which would be a better choice, I think.

File: backend/src/thirdParties/github/githubService.js (hunk 2 above)
Comment (harish-aka-shivi) [reply backend/src/thirdParties/github/githubService.js]: This is a sample format of the event we will receive from Github when listening to star events. We need this to show the suggestion to user when creating Zapp.

Comment (harish-aka-shivi) [reply backend/src/thirdParties/github/githubService.js]: Also, this API is used to determine whether a `Codertocat` follows a any other user.

File: backend/src/config/index.js [hunk 3]
```diff
@@ -7,9 +7,11 @@ dotenv.config({ path: devEnvPath, debug: true });
 const { DATABASE_URL } = process.env;
```
Comment (rishabh1403) [inline backend/src/config/index.js]: why 3 different de-structure?

File: backend/src/controllers/serviceControllers.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import githubService from '../thirdParties/github/githubService';
+
+const getAppropriateService = serviceName => {
```
Comment (rishabh1403) [inline backend/src/controllers/serviceControllers.js]: well the if/else makes no sense here hahah
it's returning the same thing eitherwise

File: backend/src/controllers/serviceControllers.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  if (serviceName === 'github') {
+    return githubService;
+  }
+  return githubService;
+};
+
+const getServices = async (req, res) => {
```
Comment (rishabh1403) [inline backend/src/controllers/serviceControllers.js]: async function without await?

Try/ catch makes no sense here. There's an assignment and return. Which one do you think would throw ?

File: backend/src/controllers/serviceControllers.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    res.send({ services });
+  } catch (error) {
+    res.status(500).send();
+  }
+};
+
+// only supporting github for now
+const getActionsForService = async (req, res) => {
+  try {
+    const service = getAppropriateService(req.param('service'));
```
Comment (rishabh1403) [inline backend/src/controllers/serviceControllers.js]: `req.param` is a function ??

File: backend/src/controllers/serviceControllers.js (hunk 1 above)
Comment (rishabh1403) [reply backend/src/controllers/serviceControllers.js]: yes, please change accordingly

File: backend/src/controllers/serviceControllers.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import githubService from '../thirdParties/github/githubService';
+
+// we are supporting one service now, but in
+// the future we can add more services.
+const getAppropriateService = serviceName => {
+  switch (serviceName) {
```
Comment (rishabh1403) [inline backend/src/controllers/serviceControllers.js]: Still it's the same...no matter what the condition is, you are returning githubService. Which condition are you testing

**Pull request merged**